### PR TITLE
feat: 添加西南石油大学财经学院和信息学院

### PR DESCRIPTION
**Pull request by rivertwilight**
## 该 PR 相关 Issue / Involved issue

Close #

## 完整路由地址 / Example for the proposed route(s)

```routes
/swpu/is/xyxx
/swpu/is/tzgg
/swpu/is/xsgz
/swpu/is/jyjx
/swpu/is/zsjy
/swpu/cjxy/xyxw
/swpu/cjxy/xytz
```

## 新 RSS 检查列表 / New RSS Script Checklist

- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [ ] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [x] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/v2/swpu/is.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    title = title.substring(0, title.indexOf('-'));
+
+    const items = [];
+    $('tr[height="20"]').each((i, elem) => {
+        items.push({
+            title: $('a[title]', elem).text().trim(),
+            pubDate: timezone(parseDate($('td:eq(1)', elem).text(), 'YYYY年MM月DD日'), +8),
+            link: `https://www.swpu.edu.cn/is/${$('a[href]', elem).attr('href').split('../')[1]}`,
+        });
+    });
```
Comment (TonyRL) [inline lib/v2/swpu/is.js:22]:
```suggestion
    const items = $('tr[height="20"]')
        .toArray()
        .map((elem) => ({
            title: $('a[title]', elem).text().trim(),
            pubDate: timezone(parseDate($('span[class^=timestyle]', elem).text(), 'YYYY/MM/DD'), +8),
            link: `https://www.swpu.edu.cn/is/${$('a[href]', elem).attr('href').split('../')[1]}`,
        }));
```

File: lib/v2/swpu/is.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                    if ($(elem).css('text-align') === 'right') {
+                        item.author = $(elem).text();
+                        break;
+                    }
+                }
+            }
+
+            return item;
+        })
+    );
```
Comment (TonyRL) [inline lib/v2/swpu/is.js:47]:
```suggestion
    const out = await Promise.all(
        items.map((item) =>
            ctx.cache.tryGet(item.link, async () => {
                const res = await got(item.link);
                const $ = cheerio.load(res.data);

                if ($('title').text().startsWith('系统提示')) {
                    item.author = '系统';
                    item.description = '无权访问';
                } else {
                    item.author = '信息学院';
                    item.description = $('.v_news_content').html();
                    for (const elem of $('.v_news_content p')) {
                        if ($(elem).css('text-align') === 'right') {
                            item.author = $(elem).text();
                            break;
                        }
                    }
                }

                return item;
            })
        )
    );
```

File: lib/v2/swpu/maintainer.js [hunk 3]
```diff
@@ -3,4 +3,6 @@ module.exports = {
     '/dean/:code': ['CYTMWIA'],
     '/dxy/:code': ['CYTMWIA'],
     '/scs/:code': ['CYTMWIA'],
+    '/cjxy/:code': ['RiverTwilight'],
```
Comment (TonyRL) [inline lib/v2/swpu/maintainer.js:6]: Please insert new entries in alphabetical order

**Comment by RiverTwilight** [comment]
@TonyRL Thanks for your review! Now I've finished the lints as you suggested, could you please have a review again?

BTW, The lint check bot still throws errors but I'm sure I don't have any improvement room because of this:

image

File: lib/v2/swpu/is.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+    let title = $('title').text();
+    title = title.substring(0, title.indexOf('-'));
+
+    const items = $('tr[height="20"]')
+        .toArray()
+        .map((elem) => ({
+            title: $('a[title]', elem).text().trim(),
+            pubDate: timezone(parseDate($('td:eq(1)', elem).text(), 'YYYY年MM月DD日'), +8),
+            link: `https://www.swpu.edu.cn/is/${$('a[href]', elem).attr('href').split('../')[1]}`;
```
Comment (TonyRL) [inline lib/v2/swpu/is.js:20]:
```suggestion
            link: `https://www.swpu.edu.cn/is/${$('a[href]', elem).attr('href').split('../')[1]}`,
```

Comment (RiverTwilight) [reply lib/v2/swpu/is.js]: Done.

**Pull request merged**